sklearn/inspection/_plot/decision_boundary.py

```diff
@@ -195,6 +215,12 @@ def plot(self, plot_method="contourf", ax=None, xlabel=None, ylabel=None, **kwar
         ylabel : str, default=None
             Overwrite the y-axis label.
 
+        response_method_used : str, default=None
+            If provided, it will be used to create the plot's title.
+
+        estimator_name : str, default=None
+            If provided, it will be used to create the plot's title.
```

Comment:
Let's also add that if `None`, we try and use `name` passed at initialization, and if that is also None, we don't add a title (or title would just include response method, though not sure about that)